corehq/apps/domain/tests/test_redirect_url.py

```diff
@@ -112,65 +110,52 @@ def _call_redirect_url(*args, **kwargs):
 
 class TestCheckDomainMigration(TestCase):
     """
-    Tests the ``receiver_post`` view, which is wrapped with the
-    ``corehq.apps.domain.decorators.check_domain_migration`` decorator.
+    Tests the ``download_odk_media_profile`` view, which is wrapped with
+    the ``corehq.apps.app_manager.decorators.check_redirect`` decorator.
 
-    All relevant views are protected by that decorator during and after
-    data migration. These tests verify that the decorator returns a 302
-    redirect response when the domain's ``redirect_url`` is set, and a
-    503 service unavailable response when it is not set.
+    All relevant views are protected by that decorator. These tests
+    verify that the decorator returns a 302 redirect response when the
+    domain's ``redirect_url`` is set, and a normal response when it is
+    not set.
     """
 
     @classmethod
     def setUpClass(cls):
         super().setUpClass()
         cls.domain_obj = create_domain(DOMAIN)
-        cls.user = WebUser.create(
-            None, 'admin', 'Passw0rd!',
-            None, None,
-        )
-        cls.user.add_domain_membership(DOMAIN, is_admin=True)
-        cls.user.save()
+        cls.app = Application.new_app(DOMAIN, "TestApp")
+        cls.app.save()
         cls.client = Client()
-        cls.client.login(username='admin', password='Passw0rd!')
 
     @classmethod
     def tearDownClass(cls):
-        cls.user.delete(DOMAIN, deleted_by=None)
         cls.domain_obj.delete()
         super().tearDownClass()
 
-    @flag_enabled('DATA_MIGRATION')
     def test_redirect_response(self):
         with _set_redirect_url():
-            response = self._submit_form()
+            response = self.client.get(reverse(
+                'download_odk_media_profile',
+                args=[DOMAIN, self.app._id],
+            ), {'latest': 'true'})
             self.assertEqual(response.status_code, 302)
             self.assertEqual(
                 response.url,
-                'https://example.com/a/test-redirect-url/receiver/'
+                f'https://example.com/a/{DOMAIN}/apps/download/{self.app._id}'
+                '/media_profile.ccpr?latest=true'
             )
 
-    @flag_enabled('DATA_MIGRATION')
-    def test_service_unavailable_response(self):
```

Comment:
Is this test no longer relevant because requests like the one it was testing are now redirected?

Reply:
Good question. https://github.com/dimagi/commcare-hq/pull/34505/commits/88b8b676dcb66d4403f937fbe652e52cd8ddbbd4